### ISSUE TITLE
[Support]: Frigate hangs when started automatically after reboot

### ISSUE THREAD

**Issue by ronisaacson** [blakeblackshear/frigate#7041]
### Describe the problem you are having

I start frigate with docker-compose (my config is in #6769). Everything works fine except when I reboot the host and the container starts up automatically. Then it hangs with the log seen below. This is the complete logfile, nothing else is logged after that, and it doesn't fully shutdown to get automatically restarted, it just stays in this state.

If I `docker compose restart` after a few minutes then it comes up just fine. I assume this is hanging due to some other startup-time dependency but there's nothing else logged. Any suggestions, or anything I can do for troubleshooting while it's like this?

### Version

0.12.1-367D724

### Frigate config file

```yaml
n/a
```

### Relevant log output

```shell
Jul  4 19:17:29 host frigate[1940]: s6-rc: info: service s6rc-fdholder: starting
Jul  4 19:17:29 host frigate[1940]: s6-rc: info: service s6rc-oneshot-runner: starting
Jul  4 19:17:29 host frigate[1940]: s6-rc: info: service s6rc-fdholder successfully started
Jul  4 19:17:29 host frigate[1940]: s6-rc: info: service s6rc-oneshot-runner successfully started
Jul  4 19:17:29 host frigate[1940]: s6-rc: info: service fix-attrs: starting
Jul  4 19:17:30 host frigate[1940]: s6-rc: info: service fix-attrs successfully started
Jul  4 19:17:30 host frigate[1940]: s6-rc: info: service legacy-cont-init: starting
Jul  4 19:17:30 host frigate[1940]: s6-rc: info: service legacy-cont-init successfully started
Jul  4 19:17:30 host frigate[1940]: s6-rc: info: service log-prepare: starting
Jul  4 19:17:31 host frigate[1940]: s6-rc: info: service log-prepare successfully started
Jul  4 19:17:31 host frigate[1940]: s6-rc: info: service nginx-log: starting
Jul  4 19:17:31 host frigate[1940]: s6-rc: info: service go2rtc-log: starting
Jul  4 19:17:31 host frigate[1940]: s6-rc: info: service frigate-log: starting
Jul  4 19:17:31 host frigate[1940]: s6-rc: info: service nginx-log successfully started
Jul  4 19:17:31 host frigate[1940]: s6-rc: info: service go2rtc-log successfully started
Jul  4 19:17:31 host frigate[1940]: s6-rc: info: service go2rtc: starting
Jul  4 19:17:31 host frigate[1940]: s6-rc: info: service frigate-log successfully started
Jul  4 19:17:33 host frigate[1940]: 2023-07-04 19:17:33.735789423  [INFO] Preparing go2rtc config...
Jul  4 19:17:34 host frigate[1940]: s6-rc: fatal: timed out
Jul  4 19:17:34 host frigate[1940]: s6-svlisten1: fatal: timed out
Jul  4 19:17:55 host frigate[1940]: 2023-07-04 19:17:55.098259763  [INFO] Starting go2rtc...
Jul  4 19:17:56 host frigate[1940]: 2023-07-04 19:17:56.443788253  19:17:56.443 INF go2rtc version 1.2.0 linux/amd64
Jul  4 19:17:56 host frigate[1940]: 2023-07-04 19:17:56.444750447  19:17:56.444 INF [api] listen addr=:1984
Jul  4 19:17:56 host frigate[1940]: 2023-07-04 19:17:56.445593361  19:17:56.445 INF [rtsp] listen addr=:8554
Jul  4 19:17:56 host frigate[1940]: 2023-07-04 19:17:56.446438839  19:17:56.446 INF [srtp] listen addr=:8443
Jul  4 19:17:56 host frigate[1940]: 2023-07-04 19:17:56.447241104  19:17:56.447 INF [webrtc] listen addr=:8555
```

### FFprobe output from your camera

```shell
n/a
```

### Operating system

Other Linux

### Install method

Docker Compose

### Coral version

USB

### Network connection

Wired

### Camera make and model

n/a

### Any other information that may be helpful

Fedora 37 is the host OS

**Comment by NickM-27** [comment]
What is your docker compose config

**Comment by ronisaacson** [comment]
Same as #6769 but here it is again :

```
version: "3.9"
services:
  frigate:
    image: ghcr.io/blakeblackshear/frigate:stable
    container_name: frigate
    restart: unless-stopped
    privileged: true
    shm_size: "128mb"
    devices:
      - /dev/bus/usb:/dev/bus/usb
    volumes:
      - /etc/localtime:/etc/localtime:ro
      - .../config.yml:/config/config.yml
      - .../frigate:/media/frigate
      - type: tmpfs
        target: /tmp/cache
        tmpfs:
          size: 1000000000
    ports:
      - "5000:5000"
      - "8554:8554" # RTSP feeds
      - "8555:8555/tcp" # WebRTC over tcp
      - "8555:8555/udp" # WebRTC over udp
    environment:
      FRIGATE_RTSP_PASSWORD: "..."
    logging:
      driver: syslog
      options:
        tag: "frigate"
    labels:
      - "com.centurylinklabs.watchtower.scope=frigate"
  watchtower:
    image: containrrr/watchtower
    container_name: watchtower-frigate
    restart: unless-stopped
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock
    command: --scope frigate
    network_mode: "host"
    environment:
      WATCHTOWER_CLEANUP: "true"
      WATCHTOWER_NOTIFICATIONS: email
      WATCHTOWER_NOTIFICATION_EMAIL_FROM: ...
      WATCHTOWER_NOTIFICATION_EMAIL_TO: ...
      WATCHTOWER_NOTIFICATION_EMAIL_SERVER: 127.0.0.1
      WATCHTOWER_NOTIFICATION_EMAIL_SUBJECTTAG: frigate
    logging:
      driver: syslog
      options:
        tag: "watchtower-frigate"
    labels:
      - "com.centurylinklabs.watchtower.scope=frigate"
```

**Comment by NickM-27** [comment]
what is ` - .../frigate:/media/frigate` ? does that point to a NAS or other storage that gets mounted?

**Comment by ronisaacson** [comment]
It's a local USB. Good thought but I don't think that's the issue. In the full startup log from that run, the filesystem was mounted at 19:15:47. dockerd didn't start until 19:16:43, the container was launched at 19:17:17 and the log output begins at 19:17:29. So the filesystem was already mounted a good 90 seconds before frigate could have started looking for it.

**Comment by ronisaacson** [comment]
Hi @NickM-27, just bubbling this back up since it happened again today. The complete log output from frigate is below. The filesystems were all mounted by 18:47:46 so nothing should have been unavailable. Any suggestions on how to troubleshoot this?

```
Jul 30 18:50:03 host frigate[1856]: s6-rc: info: service s6rc-fdholder: starting
Jul 30 18:50:03 host frigate[1856]: s6-rc: info: service s6rc-oneshot-runner: starting
Jul 30 18:50:04 host frigate[1856]: s6-rc: info: service s6rc-fdholder successfully started
Jul 30 18:50:04 host frigate[1856]: s6-rc: info: service s6rc-oneshot-runner successfully started
Jul 30 18:50:04 host frigate[1856]: s6-rc: info: service fix-attrs: starting
Jul 30 18:50:04 host frigate[1856]: s6-rc: info: service fix-attrs successfully started
Jul 30 18:50:04 host frigate[1856]: s6-rc: info: service legacy-cont-init: starting
Jul 30 18:50:05 host frigate[1856]: s6-rc: info: service legacy-cont-init successfully started
Jul 30 18:50:05 host frigate[1856]: s6-rc: info: service log-prepare: starting
Jul 30 18:50:07 host frigate[1856]: s6-rc: fatal: timed out
Jul 30 18:50:07 host frigate[1856]: s6-sudoc: fatal: unable to get exit status from server: Operation timed out
```

**Comment by blakeblackshear** [comment]
If you can consistently reproduce this when restarting the host, then I would try removing volume mounts just to see if it's related even though it's technically mounted already.

My usual strategy is to start eliminating variables to narrow it down, but you have to be able to reproduce it.

**Comment by knoffelcut** [comment]
@ronisaacson I have exactly the same issue, and the impact is quite large since I don't have stable electricity supply at the moment, so constant reboots is a reality.

I can confirm that:

- I've had this issue since I started using Frigate around may (so before the big docker refactor). I'm still experiencing it on the newest version.
- I experienced it with the frigate-tensorrt containers as well as a slightly modified base container (just added ONNX support).
- `/media/frigate` is mounted to another local drive drive.

As a stopgap I added the following cronjob, it might help you. But I really want to find and fix the source of the issue
```
@reboot bash -c 'while [ ! "$(docker ps -a | grep frigate)" ]; do sleep 1; done && sleep 16 && cd <frigate docker-compose directory> && (docker compose -f docker-compose.yml stop frigate || :) && docker compose -f docker-compose.yml up frigate -d'
``` 

On this system I have 11 other containers that starts perfectly fine.
